### PR TITLE
REFACTOR: Service Settings Panel

### PR DESCRIPTION
**Pull request by NeoPlays**
- [x] Handle service settings actions in JS Backend

### CODE REVIEW

**Comment by NeoPlays** [comment]
inspired by: [stereum 1](https://github.com/stereum-dev/ethereum2-ansible/blob/main/roles/start-geth-prune/tasks/main.yaml)

`geth-prune` container logs this:
```
INFO [09-19|11:40:00.671] Maximum peer count                       ETH=50 LES=0 total=50
INFO [09-19|11:40:00.672] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [09-19|11:40:00.674] Set global gas cap                       cap=50,000,000
INFO [09-19|11:40:00.674] Allocated cache and file handles         database=/opt/data/geth/geth/chaindata cache=512.00MiB handles=524,288
INFO [09-19|11:40:00.715] Opened ancient database                  database=/opt/data/geth/geth/chaindata/ancient/chain readonly=false
INFO [09-19|11:40:00.758] Initialized state bloom                  size=2.00GiB
ERROR[09-19|11:40:00.759] Failed to prune state                    err="snapshot not old enough yet: need 128 more blocks"
snapshot not old enough yet: need 128 more blocks
```
The error is clear (Geth not synced)

For the test i'd just check if geth stopped and that the `geth-prune` container was started.
Or should i make no tests?

Feedback wanted 😅 @stefa2k @gbayasgalan

**Comment by stefa2k** [comment]
I'd suggest to check for this regex for the test:
```
^.*Failed to prune state\s*err="snapshot not old enough yet.*
```

Everything else looks good to me.

**Comment by NeoPlays** [comment]
So geth pruning is now available. starting, resyncing, pruning is disabled while pruning.

**Review by gbayasgalan — DISMISSED** [review]
ansible lgtm 👍

**Review by maxbehzadi82 — COMMENTED** [review]
js looks good

File: launcher/src/backend/ServiceManager.js [hunk 1]
```diff
@@ -144,4 +132,24 @@ export class ServiceManager {
       return []
     })
   }
+
+  async chooseServiceAction(action, service, data){
+    switch (action) {
+      case "pruneGeth":
+        if(service.servive === "GethService"){
```
[the remark this answers is not in the document]
Comment (NeoPlays) [reply launcher/src/backend/ServiceManager.js:139]: `service` is service how it is defined in pinia

**Comment by maxbehzadi82** [comment]
look at the above before merging

File: launcher/src/backend/ServiceManager.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (NeoPlays) [reply launcher/src/backend/ServiceManager.js:139]: ohh shit 😅
good job @maxbehzadi82 🚀

**Review by maxbehzadi82 — APPROVED** [review]
lg

**Pull request merged**